### PR TITLE
[1769] Add provider details page

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
We need to kill the c# app ASAP

### Changes proposed in this pull request
Add provider details view 

### Guidance to review
Example - `/organisations/2CG/2019/details`

#After 
localhost_3000_organisations_2CG_2019_details(Laptop with MDPI screen)

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Move provider enrichments to rails](https://trello.com/c/jLfUr6Eu/1769-move-provider-enrichments-to-rails)

**Comment by fofr** [comment]
I assume if this information isn't yet published we will need some sort of "To publish this contact […]" message?

File: app/decorators/application_decorator.rb [hunk 1]
```diff
@@ -1,2 +1,39 @@
 class ApplicationDecorator < Draper::Decorator
```
Comment (fofr) [inline app/decorators/application_decorator.rb]: Should this live on ApplicationDecorator or a reusable module? (Doesn't block this PR)

**Comment by fofr** [comment]
The page is missing accredited body enrichments:
`/organisations/1CS/2019/details`

## New
Screen Shot 2019-07-15 at 11 15 45

## Old
Screen Shot 2019-07-15 at 11 15 40

**Comment by defong** [comment]
> The page is missing accredited body enrichments:
> `/organisations/1CS/2019/details`
> ## New
> 
> Screen Shot 2019-07-15 at 11 15 45
> ## Old
> 
> Screen Shot 2019-07-15 at 11 15 40

It's missing something in the backend to grab the provider_code and provider name

```sql

select distinct(acp.provider_code), acp.*
from provider as p

join course as c on 
c.provider_id = p.id

join provider as acp
c.accrediting_provider_code = acp.provider_code



```

**Comment by fofr** [comment]
(This probably means preview has a bug too)

File: app/decorators/application_decorator.rb (hunk 1 above)
Comment (dankmitchell) [reply app/decorators/application_decorator.rb]: It's shared on both courses and providers now

**Comment by dankmitchell** [comment]
@fofr @defong will raise a new ticket on trello for this bug

File: app/decorators/application_decorator.rb (hunk 1 above)
Comment (fofr) [reply app/decorators/application_decorator.rb]: Yeah, but ultimately not all decorators will need this.

Comment (dankmitchell) [reply app/decorators/application_decorator.rb]: That's true 🤷‍♂

File: app/views/providers/show.html.erb [hunk 2]
```diff
@@ -33,7 +33,11 @@
       </ul>
     <% end %>
 
-    <h2 class="govuk-heading-m"><%= govuk_link_to "About your organisation", manage_ui_url("/organisation/#{@provider.provider_code}/details") %></h2>
```
Comment (fofr) [inline app/views/providers/show.html.erb]: Maybe leave this bit out and we can merge this PR as-is?

File: app/views/providers/details.html.erb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      </dl>
+    </div>
+  </div>
+
+  <div class="govuk-grid-column-one-third">
+    <aside class="app-related">
+      <div class="govuk-!-margin-bottom-6" data-qa="provider__content-status">
+        <%= provider.status_tag %>
+      </div>
+      <p class="govuk-body">Last published on <%= l(provider.last_published_at&.to_date, format: :long) %></p>
```
Comment (misaka) [inline app/views/providers/details.html.erb]: If `last_published_at` is nil, then this blows up (tested locally).

**Comment by misaka** [comment]
> The page is missing accredited body enrichments:

Good catch. Let's ticket his up separately.

**Comment by misaka** [comment]
@dankmitchell Can we actually remove the sidebar stuff from this PR and do it with a separate ticket? Just because of the logic behind published/not published, etc. We're starting to ticket up the provider enrichment stuff now anyway.

**Pull request merged**